src/hooks/useScrollToAnchor.ts

```diff
@@ -1,22 +1,30 @@
 import { useEffect } from 'preact/hooks'
 
+interface ScrollToAnchorProps {
+  offset?: number
+  trigger?: boolean
+  anchor?: string
+  callback?: FlashingCallback
+}
+
 type FlashingCallback = (
   elementToScroll?: HTMLAnchorElement,
   elementPosition?: number
 ) => void
 
-export const getHashElement = () =>
-  window.location.hash
+export const getHashElement = (anchor?: string) =>
+  anchor
```

Comment:
Logic seems pretty complicated, we return `undefined` in many cases, maybe we can simplify it?

Reply:
Returned it to `window.location.hash`. About `undefined`, it returns if there is no hash in the link or if it couldn't find a needed element. Pretty simple I guess.